lib/README.md

```diff
@@ -0,0 +1,169 @@
+[![GitHub tag (latest SemVer)](https://img.shields.io/github/tag/sifive/wit.svg?label=release)](https://github.com/sifive/wit/releases/latest)
+[![Build Status](https://travis-ci.com/sifive/wit.svg?branch=master)](https://travis-ci.com/sifive/wit)
+[![Github](https://img.shields.io/github/license/sifive/wit.svg?color=blue&style=flat-square)](LICENSE)
+
+# Wit
+Workspace Integration Tool
+
+## What is this?
+Wit is a tool for managing workspaces. It is intended as a supplement to (not a replacement for) Git-based workflows.
+The goal of wit is to enable development in a multiple user and repository environment.
+
+A wit workspace is composed of one or more packages. A package is a git repository.
+Each package may optionally contain a wit-manifest.json file which defines other packages upon which it depends.
+Wit resolves this hierarchy of dependencies and generates a flattened directory structure in which each package
+may exist only once.
+
+## How does Wit deduplicate packages?
+When multiple versions of the same package are requested, Wit chooses the latest requested version, making sure the selected version's commit is a descendant of every other requested versions' commits.
+
+## Getting started
+
+The best way to learn wit is to check out the [tutorial](share/doc/wit/tutorial.md).
+
+### Installation
+
+Wit is intended to be as lightweight as possible. It simply requires `git` and `Python` version 3.5 or greater.
+
+You can simply clone the repo and add it to your `PATH`, for example:
+```bash
+git clone https://github.com/sifive/wit.git
+PATH=$PATH:$PWD/wit
+which wit
+```
+
+It also includes a `Makefile` for installing specific versions. The installation flow requires `make`, `rsync`, and `sed`.
+
+```bash
+make install PREFIX=/path/to/installation
+```
+
+The Makefile will create a directory with the version (it even works for commits between tags)
+and copy the contents of contents of the local clone excluding the tests and metadata.
+
+### Installation with pip
+Wit is not currently registered with [pypi](https://pypi.org), however it can be installed directly from Github using pip:
+
+```bash
+pip install -U git+https://github.com/sifive/wit@master#egg=wit&subdirectory=lib
```

Comment:
Please don't encourage people to install from `master`. We've put in effort to actually do versioned releases of Wit, so I'd much rather people install from a tagged version.

If the name [`wit` were not already taken on PyPI](https://pypi.org/project/wit/), I'd also just recommend that we publish directly to PyPI. We might want to register a name like `wit-sifive` on PyPI so that people can install from there instead of pointing directly to GitHub.

Reply:
I also like the publish-to-PyPI, we could then remove the current `make install`